### ISSUE TITLE
[bug] prefer-destructuring-assignment false positive

### ISSUE THREAD

**Issue by ha1fstack** [Rel1cx/eslint-react#1332]
### Describe the bug

<img width="763" height="156" alt="Image" src="https://github.com/user-attachments/assets/652f5bbb-e537-4648-95cc-b5087c25ee97" />

forwardRef's ref should be excluded from check

### Platform and versions

```Text
node 22
2.3.9
```

### Stack trace

```Text

```
